### PR TITLE
PR for Monsoon plugin and other maintenance changes

### PR DESCRIPTION
**Pull request by EricZielinski**
1. Monsoon plugin, README, associated modules
2. Changes to main directory README to account for **clear_cache** and **device_settings_reqs** fields
3. Addition of Physalia into TravisCI coverage tests for `test_plugins.py`
4. **Clear_cache** field in `Experiment.py `for native and web experiments and any related downstream changes to Native and Web modules.  Previously, web experiments were fixed to always clear the cache.  The other reason this was added was because applications would be cached in memory after every run with native experiments.  This fact shouldn't affect energy consumption; it had more to do with making sure each subsequent run of the same application was exactly the same (ie, the first run of application X wouldn't be cached).
5. New methods in `Device.py` and `Adb.py` to allow for programmatically enabling and disabling settings on test devices
6. Unit tests for the associated changes.  Unable to integrate unit tests for Monsoon in `test_plugins.py`.  This would require a Monsoon Power Monitor, a Monsoon-configured phone since the power_meter import statement in `Monsoon.py` kickstarts a Monsoon connection process; in addition, **pytest** doesn't recognize `sudo`, which is what I need to make that connection happen.

### CODE REVIEW

**Comment by EricZielinski** [comment]
Ivano, the one failing item relates to a unit test in test_devices.py where two applications have 3 settings (1 app has 2 settings, 1 app has only 1 setting) that need to be programmatically enabled/disabled.  It tests those two apps, which means the method in configure_settings method will be called 3 times total. Then I added another app with no settings requirement and also tested "None" in case that field was entirely absent from the config.json and confirmed that the method wasn't called for these two cases.

File: tests/unit/test_devices.py [hunk 1]
```diff
@@ -47,8 +61,25 @@ def test_init(self, adb_connect):
...
+        logger.assert_called_with('Enabling setting_1')
+        configure_settings.assert_called_with(device_with_app_settings.id, "setting_1",True)
+
+        device_with_app_settings.configure_settings_device("app2")
+        logger.assert_called_with('Enabling setting_2')
+
+        device_with_app_settings.configure_settings_device(None)
+        device_with_app_settings.configure_settings_device("app3")
+
+        configure_settings.call_count == 3
```
Comment (iivanoo) [inline tests/unit/test_devices.py:81]: shouldn't this line of code be something like `assert configure_settings.call_count == 3`?

Comment (EricZielinski) [reply tests/unit/test_devices.py]: Indeed.  Whoops.
